raster-foundry/raster-foundry
# Add layer analyses histogram component and UI

## Overview

Paired with @Lknechtli on this one. 

~Marked as WIP since it needs some styling and code clean-ups. More to come once all done.~

This PR:
- adds three components: `dataVizHistogram`, `dataVizStatistics`, `histogramBar`
- updates analysis data viz UI to include the histograms


### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo

![Screen Shot 2019-03-11 at 7 34 23 PM](https://user-images.githubusercontent.com/16109558/54165085-9aa57180-4435-11e9-884f-331184f01b30.png)


## Testing Instructions

 * On lab template UI, create some templates with `This template can be run with only one project` checked
 * Go to a project's v2 UI
 * Add some layers
 * Define layer AOIs **(important)**
 * Add some scenes to these layers
 * Create some analyses with these new layers using the template(s) created from step one
 * On the analyses list UI, select multiple analyses (no more than three, see notes section), click on `Visualize` (if more than three, this button is disabled)
 * Make sure the above step takes you to analyses visualization page and:
     - map tiles load
     - histograms show up
     - stats show up

Closes #4685
